assets/js/tracking.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Function for registering tracking to a link. To add tracking to a link you
+ * need to have an "id" attritube on the <a> element and the "track-click" attribute.
+ *
+ *   - ex. <a id="get-started-no-yaml" class="btn" href="/docs/get-started" track-click>GET STARTED</a>
+ */
+$(document).ready(function() {
+
+    // Check if the analytics object and track function are available. If they are
+    // not we do not even want to attempt to track anything.
+    if (window && window.analytics && typeof window.analytics.track === "function") {
+
+        // Find all the links with a "track-click" attribute.
+        const buttons = $("a[track-click]");
+
+        function registerTracker(element) {
+
+            // Create the tracking object.
+            const trackingData = {
+                // The id of the element.
+                id: $(element).attr("id"),
+                // The destination url of the link.
+                destinationPath: $(element).attr("href"),
+                // The current path.
+                url: window.location.pathname,
+            }
+
+            // Register a listener to the link to send data to Segment
+            // when it has been clicked.
+            $(element).on("click", function(e) {
+                window.analytics.track("link-click", trackingData);
```

Comment:
Dumb question: on the click event the browser is navigating away... so will this track event actually go through? I assume it "just works", or else Segment would recommend we use something more complicated. (e.g. blocking the event from propagating, adding a 1ms timer callback, and _then_ navigating away.)

Reply:
Jquery is doing the work here for us. The `on` listener will run the code inside the block then do whatever the default functionality of the element click is. We could prevent the link from actually navigating anywhere by calling `e.preventDefault();` in the block (I'm not suggesting we do that, I just think that helps explain how jquery handles this).